addon/components/frost-modal-binding.js

```diff
@@ -1,8 +1,18 @@
 import Ember from 'ember'
 const {Component, inject, run} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
 import layout from '../templates/components/frost-modal-binding'
 import PropTypesMixin, {PropTypes} from 'ember-prop-types'
 
+/**
+ * Helper to ensure we don't try to resolve something that isn't a then-able
+ * @param {Object} thing - probably a promise
+ * @returns {Boolean} whether thing is a thenable
+ */
+function isThenable (thing) {
+  return (thing !== undefined) && (typeof thing.then === 'function')
```

Comment:
There should be a check added in between these two for the existence of `thing.then`

Reply:
If `thing` is not defined, then `thing.then` will simply resolve as `undefined`, and typeof that will still work:

```js
> thing = {wat: 13}
{wat: 13}

> typeof(thing.then)
"undefined"
```

Reply:
Oh yeah, right.  Thanks :-)